_rules/text-contrast-afw4f7.md

```diff
@@ -0,0 +1,366 @@
+
+```html
+<p style="color: #AAA; background: linear-gradient(to right, #FFF, #00F); width: 300px">
+	Some text in English
+</p>
+```
+
+#### Failed Example 3
+
+This text has insufficient contrast with the background image.
+
+```html
+<p
+	style="color: #555; height:50px; padding-top:25px; background: black no-repeat -20px -20px url('../test-assets/contrast/black-hole.jpeg');"
+>
+	Black hole sun
+</p>
+```
+
+#### Failed Example 4
+
+This text fails because the alpha transparency significantly lowers the contrast of the otherwise black text.
+
+```html
+<p style="color: rgba(0,0,0,.3); background: #FFF">
+	Some text in English
+</p>
+```
+
+#### Failed Example 5
+
+This text fails because the CSS opacity property significantly lowers the contrast of the otherwise black text.
+
+```html
+<div style="background: #FFF">
+	<p style="color: #000; opacity: .3">
+		Some text in English
+	</p>
+</div>
+```
+
+#### Failed Example 6
+
+Even though the text is in a shadow DOM tree, the text is part of a child [text node](https://dom.spec.whatwg.org/#text) of an HTML element in the flat tree and has insufficient contrast
+
+```html
+<p style="color: #aaa; background: #fff;" id="p"></p>
+<script>
+	const shadowRoot = document.getElementById('p').attachShadow({ mode: 'open' })
+	shadowRoot.textContent = 'Some text in English'
+</script>
+```
+
+#### Failed Example 7
+
+The text is placed over a black and white background. The grey text fails because of its opacity, where the text on the black background becomes too dark, and the text on the white background becomes too light.
+
+```html
+<style>
+	#backgroundSplit {
+		color: rgba(90, 90, 90, 0.8);
+		background-position: top 0 left 0;
+		background-image: linear-gradient(90deg, transparent, transparent 3.3em, black 3.3em, black 6em);
+		padding: 0 1em;
+	}
+</style>
+<span id="backgroundSplit">
+	Hello world
+</span>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+This is invisible text.
+
+```html
+<p style="display: none">Some invisible text in English</p>
+```
+
+#### Inapplicable Example 2
+
+The text is inapplicable because it is positioned off screen, hence not [visible](#visible).
+
+```html
+<p style="position:absolute; top: -999em">Some invisible text in English</p>
+```
+
+#### Inapplicable Example 3
+
+This text has the same foreground and background colors, hence it is not [visible](#visible).
+
+```html
+<p style="color: white; background: white;">Some white on white text in English</p>
+```
+
+#### Inapplicable Example 4
+
+This text is not the child of an HTML element.
+
+```html
+<svg>
+	<text x="0" y="15">I love SVG!</text>
+</svg>
+```
+
+#### Inapplicable Example 5
+
+This is not a [text node](https://dom.spec.whatwg.org/#text) but an image.
+
+```html
+<p>
+	<img scr="../test-assets/contrast/example.png" alt="example" />
+</p>
+```
+
+#### Inapplicable Example 6
+
+This text is contained in a native button.
```

Comment:
I think description needs to point out that `button` elements have an implicit role of `button` which inherits from `widget`.
+ link to def of the `button` mapping in the AAM (looks like this is wanted in #1063 and we should do the correct thing from the start if possible…)